packages/agent-toolkit/src/core/tools/platform-api-tools/move-object-tool/move-object-tool.ts

```diff
@@ -0,0 +1,119 @@
+import { z } from 'zod';
+import { updateFolder } from '../update-folder-tool/update-folder-tool.graphql';
+import { updateBoardHierarchy } from './update-board-hierarchy.graphql';
+import { updateOverviewHierarchy } from './update-overview-hierarchy.graphql';
+import { ToolInputType, ToolOutputType, ToolType } from '../../../tool';
+import { BaseMondayApiTool, createMondayApiAnnotations } from '../base-monday-api-tool';
+import { ObjectType, UpdateOverviewHierarchyMutation } from 'src/monday-graphql/generated/graphql';
+import { UpdateBoardHierarchyMutation, UpdateFolderMutation } from 'src/monday-graphql/generated/graphql';
+
+export const moveObjectToolSchema = {
+  objectType: z.nativeEnum(ObjectType).describe('The type of object to move'),
+  id: z.string().describe('The ID of the object to move'),
+  position: z
+    .object({
+      object_id: z.string().describe('The ID of the object to position the object relative to'),
+      object_type: z.nativeEnum(ObjectType).describe('The type of object to position the object relative to'),
+      is_after: z.boolean().optional().describe('Whether to position the object after the object'),
+    })
+    .optional()
+    .describe('The new position of the object'),
+  parentFolderId: z.string().optional().describe('The ID of the new parent folder'),
+  workspaceId: z.string().optional().describe('The ID of the workspace containing the object'),
```

Comment:
If I understand correcly, not all are needed all the time. can we update the description of the tool + the description of the argumanets to tell when should we use which params?